spec/services/join_course_spec.rb

```diff
@@ -23,6 +23,23 @@
     enroll_as_instructor
   end
 
+  context 'optionally with or without a real_name' do
```

Comment:
Instead of modifying the subject within the individual tests, I think it would be clearer if we create a context for 'with real name' and 'without real name', and set the subject within that context so that it's more explicit that you're basically testing `JoinCourse.new` with and without the optional param. That will also let us avoid adding the attr_accessor that isn't otherwise used.